app/src/main/java/ai/elimu/appstore/BaseApplication.java

```diff
@@ -84,4 +97,26 @@ public Retrofit getRetrofit() {
 
         return retrofit;
     }
+
+    /**
+     * Provides {@SecurePreferences} singleton instance to access shared preferences
+     *
+     * @return A single instance of {@SecurePreferences}
+     */
+    public static SharedPreferences getSharedPreferences() {
+        if (sSecurePreferences == null) {
+            synchronized (BaseApplication.class) {
+                if (sSecurePreferences == null) {
+                    sSecurePreferences = new SecurePreferences(getAppContext(), PREF_PASSWORD,
```

Comment:
Is there any way we could use a value that is not a static phrase ("appstore123") when initializing the SecurePreferences? For example by using Android's Keystore, or some other way of dynamically generating the password? https://github.com/scottyab/secure-preferences/issues/1

Reply:
One of the approaches to generate run time password is to ask user to enter his/her own password.
This way, the encryption password is not stored anywhere in the app.

Reply:
Can we use the license number as the "password"?

Reply:
Yes, sure. Will change it.

Reply:
Currently the App shared preferences is initialized in BaseApplication.onCreate(), which is first called before everything else.
Meanwhile the license number is provided by user. We only have it once we go to LicenseNumberActivity.

Therefore we haven't got the license number yet while initialize the preferences. Do you have any other idea?

Reply:
How about using UUID.randomUUID().toString(), which is a randomly unique string as the password?

Reply:
> How about using UUID.randomUUID().toString(), which is a randomly unique string as the password?

Will the value always be the same, so that it can be used for decrypting the values later?

Reply:
It looks like they are developing a solution with the Android Keystore, but the pull request has not been accepted yet and there's no documentation https://github.com/scottyab/secure-preferences/pull/58

Reply:
@tuancoltech Using a hardware-backed encryption using the Android keystore sounds like a good idea to me.